ratatui-core/src/buffer/buffer.rs

```diff
@@ -1109,12 +1110,65 @@ mod tests {
         );
     }
 
+    #[test]
+    fn merge_diff_idempotent() {
+        let mut prev = Buffer::with_lines(["123"]);
+        let next = Buffer::with_lines(["456"]);
+        prev.merge(&next);
+
+        let diff = prev.diff(&next);
+        assert_eq!(diff, []);
+    }
+
+    #[test]
+    fn merge_diff_idempotent_multi_width() {
+        let mut sequences = Buffer::empty(Rect::new(0, 0, 20, 1));
+
+        let mut kitty_image_placeholder_start = String::new();
+        // The first control sequence is to store the cursors state, including foreground color.
+        // It is necessary because we need to introduce a foreground color, but restore later to
+        // the previous foreground color, to which we don't have access in this buffer.
+        // At this point, the symbol width is already greater than one and would cause the next
+        // cell to be skipped, if we don't `set_forced_width(Some(1))`.
+        kitty_image_placeholder_start.push_str("\x1b[s");
+        // The second control sequence sets the foreground color to some 24 bit value, which is an
+        // imaginary "kitty protocol image id".
+        kitty_image_placeholder_start.push_str("\x1b[38;2;0;0;0m");
+        // The following are some special (reserved) unicode placeholder symbols.
+        // 10EEEE is the actual placeholder, and the 3 following characters would contain some row,
+        // column, and extra "image id" data.
+        kitty_image_placeholder_start.push_str("\u{10EEEE}\u{305}\u{305}\u{305}");
+
+        sequences
+            .cell_mut((0, 0))
+            .unwrap()
+            .set_symbol(&kitty_image_placeholder_start)
+            .set_diff_option(CellDiffOption::UnitWidth);
+
+        // Add two follow up placeholder symbols that have a natural width of 1.
+        sequences.cell_mut((1, 0)).unwrap().set_char('\u{10EEEE}');
+        sequences.cell_mut((2, 0)).unwrap().set_char('\u{10EEEE}');
+        // The last symbol also has a control sequence which restores the style, so we need to
+        // force the width again to 1 to avoid skipping the next cell.
+        sequences
+            .cell_mut((3, 0))
+            .unwrap()
+            .set_symbol("\u{10EEEE}\x1b[u")
+            .set_diff_option(CellDiffOption::UnitWidth);
+
+        let mut buffer = Buffer::filled(Rect::new(0, 0, 20, 1), Cell::new("x"));
+        buffer.merge(&sequences);
+
+        let diff = buffer.diff(&sequences);
```

Comment:
This test fails, if `Cell::width` returns the symbol width, instead of 1 when `diff_option` is `UnitWidth`.